graphs/possible_bipartition.py

```diff
@@ -1,12 +1,46 @@
-# Can be used for BFS
-from collections import deque 
+from collections import defaultdict  
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(V+E)
+        Space Complexity: O(V+E)
     """
-    pass
+    # create an adjacency list to represent the graph
+    graph = defaultdict(list)
+    for dislike in dislikes:
+        # if there are no enemies, no need to add to graph
+        if len(dislike) > 1:
+            dog, enemy = dislike[0], dislike[1]
```

Comment:
👀 I think there's a slight misconception here about how `dislikes` works

The dog (aka the nodes) is actually represented by the indices of `dislikes`. The list stored at each index represents the dogs that are disliked by the dog at index `i` (aka the edges)
For example if we had this:
`dislikes = [    [1,2,3], [], [3], [2]]`
It would mean that
Dog 0 dislikes dogs 1,2, and 3
Dog 1 doesn't dislike any dogs
Dog 2 dislikes dog 3
Dog 3 dislikes dog 2 back

Here, you are labeling the first in a list of disliked dogs as the actual node, and only adding the second dog in the list of disliked dogs to the node's adjacency list